### PR TITLE
Fixes game options disappearing when host is observer

### PR DESCRIPTION
**Pull request by Justify87**
Related issue #1346

### CODE REVIEW

**Comment by Justify87** [comment]
Different solution:
Host should be allowed to be observer and launch the game. Someone might
host a game just to cast it as an observer.

GetPlayerCount() only counts the players/AI that occupy a game slot

**Comment by aeoncleanse** [comment]
This doesn't seem to work. Four AIs in slots, host in observer, change map, and the options button greys out.

**Comment by Justify87** [comment]
Strange. When I tested it, It worked. I will take a look.

**Comment by aeoncleanse** [comment]
Take a look at ice branch. I just posted a working solution.

File: lua/ui/lobby/lobby.lua [hunk 1]
```diff
@@ -5368,13 +5368,16 @@ function InitHostUtils()
             -- Is at least one person not ready?
             local playerNotReady = GetPlayersNotReady() ~= false
```
Comment (ChrisKitching) [inline lua/ui/lobby/lobby.lua]: Best approach is probably to change GetPlayersNotReady() to return true when there are no players.

That'd be a consistent way to solve this whole class of problems.,

**Comment by Justify87** [comment]
@aeoncleanse 

Options get unavailable as soon as I (the host) become an observer, with no AI or Players present in any slot. Closed slots are ignored too.

**Pull request closed without merging**